Zend/zend_execute.c

```diff
@@ -1293,15 +1440,44 @@ static zend_never_inline void zend_pre_incdec_overloaded_property(zval *object,
 		}
 		ZVAL_DEREF(z);
 		SEPARATE_ZVAL_NOREF(z);
-		if (inc) {
-			increment_function(z);
+		if (ZEND_OBJECT_HAS_TYPE_HINTS(object) && (prop_info = zend_object_fetch_property_type_info(object, property, cache_slot))) {
+			zval tmp;
+
+			ZVAL_DUP(&tmp, z);
```

Comment:
why do you duplicate here? we even don't exactly know if this property has a type hint.